### PR TITLE
Fix compiler warnings and new Block() usage

### PR DESCRIPTION
**Pull request by fassadlr**
This should be a pretty easy PR to review.

We are now down to 25 compiler warnings.

### CODE REVIEW

File: src/NBitcoin.Tests/ChainTests.cs [hunk 1]
```diff
@@ -554,7 +555,7 @@ public ChainedHeader AppendBlock(ChainedHeader previous, params ConcurrentChain[
             uint nonce = RandomUtils.GetUInt32();
             foreach (ConcurrentChain chain in chains)
             {
-                Block block = TestUtils.CreateFakeBlock(this.network.CreateTransaction());
+                Block block = TestUtils.CreateFakeBlock(this.network, this.network.CreateTransaction());
```
Comment (bokobza) [inline src/NBitcoin.Tests/ChainTests.cs]: you can just call `TestUtils.CreateFakeBlock(this.network);` here.

File: src/NBitcoin.Tests/TestUtils.cs [hunk 2]
```diff
@@ -32,27 +20,28 @@ internal static byte[] ParseHex(string data)
...
             block.AddTransaction(tx);
             block.UpdateMerkleRoot();
             return block;
         }
 
-        public static Block CreateFakeBlock()
+        public static Block CreateFakeBlock(Network network)
         {
-            Block block = CreateFakeBlock(new Transaction());
+            Block block = CreateFakeBlock(network, network.CreateTransaction());
```
Comment (bokobza) [inline src/NBitcoin.Tests/TestUtils.cs]: you can just call `CreateFakeBlock(this.network);` here.

File: src/Stratis.Bitcoin.Tests/P2P/PeerConnectorTests.cs [hunk 3]
```diff
@@ -85,8 +84,8 @@ public void PeerConnectorAddNode_CanAlwaysStart()
             var nodeSettings = new NodeSettings();
 
             var connectionSettings = new ConnectionManagerSettings(nodeSettings);
-
-            var connector = new PeerConnectorAddNode(this.asyncLoopFactory, DateTimeProvider.Default, this.extendedLoggerFactory, this.network, this.networkPeerFactory, this.nodeLifetime, nodeSettings, connectionSettings, peerAddressManager, new SelfEndpointTracker());
+            var networkPeerFactory = new Mock<INetworkPeerFactory>();
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Tests/P2P/PeerConnectorTests.cs]: How come this changed?

File: src/NBitcoin.Tests/ChainTests.cs (hunk 1 above)
Comment (fassadlr) [reply src/NBitcoin.Tests/ChainTests.cs]: Fixed :)

File: src/NBitcoin.Tests/TestUtils.cs (hunk 2 above)
Comment (fassadlr) [reply src/NBitcoin.Tests/TestUtils.cs]: Not with this one, as you will end up recursively calling the same method? There is a `public static Block CreateFakeBlock(Network network, Transaction tx)` above.

File: src/Stratis.Bitcoin.Tests/P2P/PeerConnectorTests.cs (hunk 3 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin.Tests/P2P/PeerConnectorTests.cs]: There was a global this.networkPeerFactory field in the class that wasn't used, so I implemented them in each test.

File: src/NBitcoin.Tests/TestUtils.cs (hunk 2 above)
Comment (bokobza) [reply src/NBitcoin.Tests/TestUtils.cs]: ah yes lol, what was I thinking.

**Pull request merged**